ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh

```diff
@@ -260,6 +260,29 @@ echo "$(date) - waiting for clusteroperators to finish progressing..."
 oc wait clusteroperators --all --for=condition=Progressing=false --timeout=10m
 echo "$(date) - all clusteroperators are done progressing."
 
+# this works around a problem where tests fail because imagestreams aren't imported.  We see this happen for exec session.
+echo "$(date) - waiting for non-samples imagesteams to import..."
+count=0
+while :
+do
+  non_imported_imagestreams=$(oc -n openshift get is -o go-template='{{range .items}}{{$namespace := .metadata.namespace}}{{$name := .metadata.name}}{{range .status.tags}}{{if not .items}}{{$namespace}}/{{$name}}:{{.tag}}{{"\n"}}{{end}}{{end}}{{end}}')
+  if [ -z "${non_imported_imagestreams}" ]
+  then
+    echo "Trying to break loop"
+    break
+  fi
+  echo "${non_imported_imagestreams}"
+
+  count=$((count+1))
+  if (( count > 10 )); then
+    echo "Failed while waiting on imagestream import"
+    exit 1
+  fi
+
+  sleep 60
+done
+echo "$(date) - all imagestreams are imported."
```

Comment:
From: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/22359/rehearse-22359-pull-ci-openshift-installer-release-4.9-e2e-aws-upgrade/1444036535581151232/artifacts/e2e-aws-upgrade/openshift-e2e-test/build-log.txt

Fri Oct  1 21:24:47 UTC 2021 - waiting for non-samples imagesteams to import...
Trying to break loop
Fri Oct  1 21:24:48 UTC 2021 - all imagestreams are imported.

This shows just one second, presumably this is normally all done by this point we're just preventing the flakes when it isn't?

Reply:
> This shows just one second, presumably this is normally all done by this point we're just preventing the flakes when it isn't?

correct